main.sh

```diff
@@ -213,7 +267,7 @@ for tool in "${tools[@]}"; do
             retry curl --proto '=https' --tlsv1.2 -fsSL --retry 10 --retry-connrefused "${url}" \
                 | tar xzf - -C ${CARGO_HOME:-~/.cargo}/bin
             ;;
-        *) bail "unsupported tool '${tool}'" ;;
+        *) cargo_binstall "$tool" "$version" ;;
```

Comment:
I think we can also support the installation of the cargo-binstall itself. It would be useful if the user wants to use non-default cargo-binstall options. (e.g., a case where the host is linux-gnu but the user wants to install linux-musl binaries.)

I think we can probably implement this by skipping the installation of `$tool` if `$tool == cargo-binstall`.
(We do not want to allow the installation of a version of cargo-binstall that is incompatible with install-action, so it is preferable not to support version specification.)

Reply:
That's a good idea!

I think it is better to refactor `cargo_binstall` and split out a new function `install_cargo_binstall`, as it will be much cleaner than putting it in one function.

Reply:
Done!